index.d.ts

```diff
@@ -766,7 +766,7 @@ export interface DialogProps {
   /**
    * Title of the Dialog. Titles should use Title Case.
    */
-  title?: string
+  title?: string | React.ReactNode
```

Comment:
@williamgrosset `React.ReactNode` is a superset type that includes `string`, so let's just keep it as `React.ReactNode` for simplicity :) 

```suggestion
  title?: React.ReactNode
```